src/common/src/util/value_encoding/column_aware_row_encoding.rs

```diff
@@ -151,21 +220,22 @@ impl Serializer {
 
 /// Column-Aware `Deserializer` holds needed `ColumnIds` and their corresponding schema
 /// Should non-null default values be specified, a new field could be added to Deserializer
-pub struct Deserializer<'a> {
+#[derive(Clone)]
+pub struct Deserializer {
     needed_column_ids: BTreeMap<i32, usize>,
-    schema: &'a [DataType],
+    schema: Vec<DataType>,
 }
 
-impl<'a> Deserializer<'a> {
-    pub fn new(column_ids: &'a [ColumnId], schema: &'a [DataType]) -> Self {
+impl Deserializer {
+    pub fn new(column_ids: &[ColumnId], schema: &[DataType]) -> Self {
         assert_eq!(column_ids.len(), schema.len());
         Self {
             needed_column_ids: column_ids
                 .iter()
                 .enumerate()
                 .map(|(i, c)| (c.get_id(), i))
                 .collect::<BTreeMap<_, _>>(),
-            schema,
+            schema: schema.to_vec(),
```

Comment:
This will lead to allocation and might be costly if frequently called. Is it possible to stay using reference here?

Reply:
I found it hard to pass the lifetime checker if it remains as a `&[ColumnId]`, could you please suggest some alternatives?